plonky2x/src/frontend/eth/beacon/builder.rs

```diff
@@ -40,7 +45,47 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         self.add_simple_generator(&generator);
         generator.validator
     }
+
+    /// SHA256 implementation for builder 
+        pub fn sha(&mut self, input: &[ByteVariable]) -> Bytes32Variable {
+            let input_bool: Vec<BoolTarget> = input.iter().flat_map(|byte| byte.0.iter().cloned().map(|bool_variable| BoolTarget::new_unsafe(bool_variable.0 .0))).collect();
+            let hash_bool = sha256::sha256::<F,D>(&mut self.api, &input_bool);
+            let hash_bytes_vec = hash_bool.chunks(8).map(|chunk| ByteVariable(array![i => BoolVariable::from(chunk[i].target); 8])).collect::<Vec<_>>();
+            let mut hash_bytes_array = [ByteVariable::init(self); 32];
+            hash_bytes_array.copy_from_slice(&hash_bytes_vec);
+            Bytes32Variable(BytesVariable(hash_bytes_array))
+        }
+    
+    /// SSZ Merkle Proof 
+    pub fn ssz_restore_merkle_root(
+        &mut self,
+        leaf: Bytes32Variable,
+        index: u64,
+        depth: usize,
```

Comment:
I think we don't actually need depth as an argument. It can use https://github.com/succinctlabs/telepathy/blob/main/contracts/src/libraries/SimpleSerialize.sol#L30-L48 to automatically deduce the depth.